doc/specs/stdlib_logger.md

```diff
@@ -0,0 +1,934 @@
+|------------|-------------|
+| `success` | no error was detected|
+| `close_failure` | a `close` statement for an I/O unit failed|
+| `invalid_index_error` | the `column` was invalid for the given `line` |
+| `non_sequential_error` | the I/O unit did not have `SEQUENTIAL` access|
+|`open_failure` | an `open` statement failed |
+| `read_only_error` | an output unit did not have an `access` specifier of `'WRITE'` or `'READWRITE'` |
+| `unformatted_in_error` | the unit did not have a `form` of `'FORMATTED'`|
+|`unopened_in_error`| the unit was not opened|
+| `write_fault` | one of the writes to `log_units` failed|
+
+## The derived type: logger_t
+
+### Status
+
+Experimental
+
+### Description
+
+Serves to define 'logger' variables to be used in reporting
+significant events encountered during the execution of a program.
+
+### Syntax
+
+type(logger_t) :: variable
+
+### Private attributes
+
+| Attribute | Type | Description | Initial value |
+|-----------|------|-------------|---------| 
+| `add_line` | Logical | Flag to precede output with a blank line |`.true.`|
+|`indent_lines` | Logical| Flag to indent subsequent lines by four columns|`.true.`|
+|`log_units`|Integer array| List of I/O units used for output| empty|
+|`max_width`| Integer | Maximum column width of output | 80 |
+|`time_stamp`| Logical | Flag to precede output by a time stamp | `.true.` |
+|`units` | Integer | Count of the number of active output units | 0 |
+
+## The `stdlib_logger` variable
+
+The module defines one public variable, `global_logger`, of type
+`logger_t`. As might be guessed from its name, `global_logger` is
+intended to serve as the default logger for use throughout an
+application.
+
+
+### Overview of the `logger_t` methods
+
+The module defines twelve public procedures: one function and eleven
+subroutines. All are methods of the `logger_t` derived type. The
+procedures are:
+
+|Procedure|Class|Description|
+|---------|-----|-----------|
+|`add_log_file`|Subroutine|opens a file using `newunit`, and adds the resulting unit to the `log_units` list|
+|`add_log_unit`| Subroutine|adds an existing unit to the `log_units` list|
+|`configure`|Subroutine| configures the details of the logging process|
+|`configuration`|Subroutine| reports the details of the logging configuration|
+|`log_error`| Subroutine|sends a message prepended by `'ERROR: '` optionally followed by a `stat` or `errmsg`|
+|`log_information`|Subroutine| sends a message prepended by `'INFORMATION: '`|
+|`log_io_error`|Subroutine|sends a message prepended by `'I/O ERROR: '` optionally followed by an `iostat` or `iomsg`|
+|`log_message`|Subroutine| sends a message|
+|`log_text_error`|Subroutine| sends a message describing an error found in a line of text|
+|`log_units_assigned`|Function| returns the number of active I/O units in `log_units`|
+|`log_warning`|Subroutine| sends a message prepended by `'WARNING: '`|
+|`remove_log_unit`|Subroutine| removes the `unit` number from the `log_units` array|
```

Comment:
Should it be added a `remove_log_file`?

Reply:
FWIW it can be done, but I don't keep track of `log_units` by filename so the logic will be more convoluted and may not be reliable. I could do `inquire(file=filename,number=unit), but the standard doesn't guarantee that the file is connected with one unit number. I could also keep a separate list of filenames with corresponding unit numbers for files opened with `add_log_file`.